library/std/src/sys/unix/thread.rs

```diff
@@ -285,17 +285,18 @@ impl Drop for Thread {
 }
 
 #[cfg(any(target_os = "linux", target_os = "macos", target_os = "ios", target_os = "watchos"))]
-fn truncate_cstr(cstr: &CStr, max_with_nul: usize) -> crate::borrow::Cow<'_, CStr> {
-    use crate::{borrow::Cow, ffi::CString};
-
-    if cstr.to_bytes_with_nul().len() > max_with_nul {
-        let bytes = cstr.to_bytes()[..max_with_nul - 1].to_vec();
-        // SAFETY: the non-nul bytes came straight from a CStr.
-        // (CString will add the terminating nul.)
-        Cow::Owned(unsafe { CString::from_vec_unchecked(bytes) })
-    } else {
-        Cow::Borrowed(cstr)
-    }
+fn truncate_cstr<const MAX_WITH_NUL: usize>(cstr: &CStr) -> [i8; MAX_WITH_NUL] {
```

Comment:
pthread_setname_np takes a *const c_char not a *const i8. Doesn't it make more sense to use the c_char type here? (Though that may need a static assert that c_char is 1 byte?!)

Reply:
Updated to not use transmute.